daemon/cluster/convert/network.go

```diff
@@ -30,10 +31,15 @@ func networkFromGRPC(n *swarmapi.Network) types.Network {
 				Attachable:  n.Spec.Attachable,
 				Ingress:     n.Spec.Ingress,
 				IPAMOptions: ipamFromGRPC(n.Spec.IPAM),
+				Scope:       netconst.SwarmScope,
 			},
 			IPAMOptions: ipamFromGRPC(n.IPAM),
 		}
 
+		if n.Spec.ConfigFrom != "" {
```

Comment:
If we agree on the generic reference structure, I'd love to make the change in swarmkit specs.